server/src/test/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregatorTests.java

```diff
@@ -921,22 +922,28 @@ private <T> void termsAggregator(
                         .size(size)
                         .shardSize(size)
                         .field("field")
+                        .showTermDocCountError(true)
                         .order(bucketOrder);
 
-                    Terms result = searchAndReduce(
-                        indexSearcher,
-                        new AggTestConfig(aggregationBuilder, fieldType).withSplitLeavesIntoSeperateAggregators(false)
-                    );
+                    AggTestConfig aggTestConfig = new AggTestConfig(aggregationBuilder, fieldType);
+                    Terms result = searchAndReduce(indexSearcher, aggTestConfig);
                     assertEquals(size, result.getBuckets().size());
+                    HashMap<Object, Long> expectedCounts = new HashMap<>();
                     for (int i = 0; i < size; i++) {
                         Map.Entry<T, Integer> expected = expectedBuckets.get(i);
+                        Object key = (valueType == ValueType.IP) ? String.valueOf(expected.getKey()).substring(1) : expected.getKey();
+                        expectedCounts.put(key, expected.getValue().longValue());
+                    }
+                    long docCountError = result.getDocCountError();
+                    for (int i = 0; i < size; i++) {
                         Terms.Bucket actual = result.getBuckets().get(i);
-                        if (valueType == ValueType.IP) {
-                            assertEquals(String.valueOf(expected.getKey()).substring(1), actual.getKey());
-                        } else {
-                            assertEquals(expected.getKey(), actual.getKey());
+                        long expectedCount = expectedCounts.getOrDefault(actual.getKey(), 0L);
```

Comment:
We should definitely add a comment here.  This was confusing enough the first time through, I don't want to have to figure this out again next time I touch it.